### PR TITLE
[CX_CLEANUP] - Complete the dependency implementation

### LINKED ISSUE #2710: [CX_CLEANUP] - Complete the dependency implementation
### What is this task and why do we need to work on it?

- Fill in implementations (i.e., the validation part) for the three `EventDependency`s.
- Store the VID and leaf commitment for field matches among events.

### What work will need to be done to complete this task?

_No response_

### Are there any other details to include?

Design doc: https://www.notion.so/espressosys/Refactor-Voting-Logic-9afa14dd027d453aa9b4a80a52de5e44.

### What are the acceptance criteria to close this issue?

Code compiles, PR approved, test to be done next.

### Branch work will be merged to (if not the default branch)

_No response_

### PR DESCRIPTION
**Pull request by shenkeyao**
Closes #2710.

### This PR: 
* Completes the validations for the quorum proposal, DAC, and VID share.
* Updates tests.
* Renames `VidDisperseRecv` to `VIDShareRecv` to be consistent with the diagram.

### This PR does not: 
* Integrate the quorum vote task with other tasks.
* Validate the view change evidence or upgrade certificate.

### Key places to review: 
* `crates/task-impls/src/quorum_vote.rs`, especially the quorum proposal validation logic.

(Many of the other files only contain renamings.)

* `just async_std test_quorum_vote_task`

### CODE REVIEW

File: crates/task-impls/src/quorum_proposal.rs [hunk 1]
```diff
@@ -488,11 +488,6 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>> QuorumProposalTaskState<TYPE
                 if !validate_quorum_proposal(proposal.clone(), event_sender.clone()) {
                     return;
                 }
-                broadcast_event(
```
Comment (jparr721) [inline crates/task-impls/src/quorum_proposal.rs:491]: Was this causing problems for you? I have it removed in my other PR.

File: crates/testing/tests/tests_1/quorum_proposal_task.rs [hunk 2]
```diff
@@ -58,7 +59,7 @@ async fn test_quorum_proposal_task_quorum_proposal() {
             SendPayloadCommitmentAndMetadata(payload_commitment, (), ViewNumber::new(2)),
         ],
         outputs: vec![
-            exact(QuorumProposalValidated(proposals[1].data.clone())),
+            quorum_proposal_validated(),
```
Comment (jparr721) [inline crates/testing/tests/tests_1/quorum_proposal_task.rs:62]: It's interesting that this passes even though the broadcast was removed above?

**Review by jparr721 — COMMENTED** [review]
Curious about the changes to the Quorum Proposal Task

**Comment by shenkeyao** [comment]
> Curious about the changes to the Quorum Proposal Task

@jparr721 I did the test update as discussed and the proposal task tests now pass!

**Comment by shenkeyao** [comment]
I don't think the PR fully handles the case where the parent info is missing, though tests pass. I'm going to send an update.

**Comment by shenkeyao** [comment]
> I don't think the PR fully handles the case where the parent info is missing, though tests pass. I'm going to send an update.

Should be fixed now. Ready for review!

**Review by jparr721 — APPROVED** [review]

**Pull request merged**